### ISSUE TITLE
Ch. 10 and Ch. 9 seem out of order.

### ISSUE THREAD

**Issue by achernyak** [MostlyAdequate/mostly-adequate-guide#214]
According to category theory the level of complexity rises in the order of Functor -> Applicative -> Monad. Shouldn't the same order be used in the book? Especially since you can use methods from a functor to define Applicative instances and methods from both the Functor and Applicative instance to define Monad instances.

I think this is ordering is also important because the structural constraints for each of these classes is built on top of the previous structural constraint. The way the book is makes it seem like the Applicative should be defined in terms of a Monad and not the other way around.

[Fantasy Land spec for reference](https://github.com/fantasyland/fantasy-land)

**Comment by DrBoolean** [comment]
Really good points there and that's how it's normally taught.

Despite these terrific reasons, I found that was easier for people to conceptualize joining two nested functors than applying functions within them. In addition to that, monads give us the power to solve almost all the inevitable problems that arise from the use of functors. So after ch8, if a few days of work pass, it's going to address the need right off the bat. Finally, I think people have an "achievement unlocked" feeling after learning monads, which keeps the spark for learning going and I wanted to get there as early as I could.

For these reasons, I chose to teach it out of order like that.
